packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js

```diff
@@ -538,4 +546,98 @@ export default class XdslStatisticsCtrl {
         this.attenuation.loading = false;
       });
   }
+
+  /**
+   * Get errors statistic for the line
+   * @param {String} period Period to request :
+   *          - daily
+   *          - monthly
+   *          - preview
+   *          - weekly
+   *          - yearly
+   * @return {promise}
+   */
+  getErrorsStatistics(period) {
+    this.errors.loading = true;
+    return this.$q
+      .all({
+        hec: this.getLinesStatistics('error:hec', period),
+        fec: this.getLinesStatistics('error:fec', period),
+        crc: this.getLinesStatistics('error:crc', period),
+      })
+      .then((stats) => {
+        this.errors.haveSeries = !!Object.values(stats).every(
+          (stat) => stat.length,
+        );
+
+        this.errors.chart = new this.TucChartjsFactory(
```

Comment:
Could the code to generate chart + series be in a specific method to be more readable ?

Reply:
It's already done by `TucChartjsFactory`, in this code we add series and labels to display into the generated chart.